hail/python/hail/linalg/utils/misc.py

```diff
@@ -48,7 +48,7 @@ def array_windows(a, radius):
         raise ValueError("array_windows: 'a' must be 1-dimensional")
     if not (np.issubdtype(a.dtype, np.signedinteger) or np.issubdtype(a.dtype, np.floating)):
         raise ValueError(
-            f"array_windows: 'a' must be an ndarray of signed integer or float values, " f"found dtype {str(a.dtype)}"
+            f"array_windows: 'a' must be an ndarray of signed integer or float values, " f"found dtype {a.dtype!s}"
```

Comment:
according to [this](https://stackoverflow.com/questions/41630728/using-s-vs-s-to-format-a-string-in-python), it can be important to format an object to a string before the f-string formats it if the `__format__` method of the object in question differs from its `__str__` method, but i tried just making an `ndarray` in `ipython` and checking if its `dtype` had different values for the two methods, and it did not, so it seems fine to get rid of it here?

Reply:
ah interesting, TIL. Either seems fine to me then.

Reply:
removed!